packages/assets-controllers/src/TokensController.ts

```diff
@@ -1013,6 +1024,24 @@ export class TokensController extends BaseController<
     return { newAllTokens, newAllIgnoredTokens, newAllDetectedTokens };
   }
 
+  #getAddressOrSelectedAddress(address: string | undefined): string {
+    if (address) {
+      return address;
+    }
+
+    return this.#getSelectedAddress();
+  }
+
+  #isInterctingWithWallet(address: string) {
+    // If the address is not defined (or empty), we fallback to the currently selected account's address
+    const selectedAccount = this.messagingSystem.call(
+      'AccountsController:getAccount',
+      this.#selectedAccountId,
+    );
+
+    return selectedAccount?.address === address;
+  }
```

Comment:
Also, there's a typo for `isIntercting` -> `isInteracting`:

```suggestion
  #isInteractingWithWallet(address: string) {
    return this.#getSelectedAddress() === address;
  }
```